### DISCUSSION TITLE
Where can i find information on editing sales invoice pdf function?

### DISCUSSION THREAD

**Discussion by MonsterITServices** [YetiForceCompany/YetiForceCRM#16792] [How to]
HI,

Where can I find the hooks and shortcodes to change the styling and edit the pdf generated functions.

**Comment by vovpff** [comment]
Look for TextParser. You can write custom TextParser function what generate whole invice appearance

**Reply by MonsterITServices** [reply]
Where do i find that? I've been through all the settings and cant see it on the back end.

**Reply by vovpff** [reply]
You need to write php code before use it

**Reply by vovpff** [reply]
You code should be stored into app/Custom folder

**Reply by MonsterITServices** [reply]
Where are the hooks/etc located we can currently use for it?
